src/user/models.py

```diff
@@ -0,0 +1,56 @@
+import datetime
+from typing import Annotated, Optional
+
+from sqlalchemy import DateTime, Enum, Integer, String, ForeignKey, func, Boolean, Index
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column, relationship
+
+
+timestamp = Annotated[
+    datetime.datetime,
+    mapped_column(nullable=False, server_default=func.CURRENT_TIMESTAMP()),
+]
+
+
+class Base(DeclarativeBase):
+    pass
+
+
+class RoleEnum(Enum):
+    USER = "USER"
+    ADMIN = "ADMIN"
+    MODERATOR = "MODERATOR"
+
+
+class User(Base):
```

Comment:
use Text instead of String(...) with construction, in postgresql it will have higher performance